src/jwk.ts

```diff
@@ -41,14 +44,31 @@ type MandatoryJwkFields = {
 
 export type Jwk = OptionalJwkFields & MandatoryJwkFields & JsonObject;
 
+export type RsaSignatureJwk = Jwk & {
+  use: "sig";
+  kty: "RSA";
+  n: string;
+  e: string;
+};
```

Comment:
I am a little worried about "e" after reading this from the [RFC](https://www.ietf.org/rfc/rfc7517.txt)
> 9.3.  RSA Private Key Representations and Blinding
   The RSA Key blinding operation [Kocher], which is a defense against
   some timing attacks, requires all of the RSA key values "n", "e", and
   "d".  However, some RSA private key representations do not include
   the public exponent "e", but only include the modulus "n" and the
   private exponent "d".  This is true, for instance, of the Java
   RSAPrivateKeySpec API, which does not include the public exponent "e"
   as a parameter.  So as to enable RSA key blinding, such
   representations should be avoided.  For Java, the
   RSAPrivateCrtKeySpec API can be used instead.  Section 8.2.2(i) of
   the "Handbook of Applied Cryptography" [HAC] discusses how to compute
   the remaining RSA private key parameters, if needed, using only "n",
   "e", and "d".

What happen if you make a Union type and have a type guard to check to which type is validating against? Or maybe "e" is optional?

Reply:
This section from RFC7518 JWA (which builds on RFC7517 that you linked) has more specific info:

> The "e" (exponent) parameter contains the exponent value for the RSA public key.  It is represented as a Base64urlUInt-encoded value.
>
> For instance, when representing the value 65537, the octet sequence to be base64url-encoded MUST consist of the three octets [1, 0, 1]; the resulting representation for this value is "AQAB".

([https://datatracker.ietf.org/doc/html/rfc7518#section-6.3.1.2](link))

And it is stated that [these members MUST be present for RSA public keys](https://datatracker.ietf.org/doc/html/rfc7518#section-6.3.1).

Think we are good?

Reply:
It appears "e" is required for the public key, so I think we can leave this as is.